### ISSUE TITLE
Add support for element integrations to map fields to Matrix/Super Table

### ISSUE THREAD

**Issue by engram-design** [verbb/formie#684]

**Comment by chasegiunta** [comment]
@engram-design Just clarifying, this should still be doable via a custom element integration or with a Formie event?

**Comment by engram-design** [comment]
Yep, but you'll just need to sort things out manually (you can't add UI to the field mapping section).

https://verbb.io/craft-plugins/formie/docs/developers/events#the-beforeSendPayload-event

```php
use verbb\formie\events\SendIntegrationPayloadEvent;
use verbb\formie\integrations\elements\Entry;
use yii\base\Event;

Event::on(Entry::class, Entry::EVENT_BEFORE_SEND_PAYLOAD, function(SendIntegrationPayloadEvent $event) {
    $entry = $event->payload;

    $entry->setFieldValue('myMatrixField', $myMatrixData);

    $event->payload = $entry;
});
```

https://craftcms.stackexchange.com/questions/21864/how-to-save-matrix-data-programmatically-in-craft-3

**Comment by marcusgaius** [comment]
I have managed to achieve the desired effect in the following manner:
In the custom template path, I have added another line in the `fields/group/_row.html` file, and set it like so: 
```{{ hiddenInput(field.getHtmlName("#{id}[type]"), groupField.handle) }}```
In the form, I have used the group field to mimic the matrix wrapper. Handles must match for all relevant structures, of course.
Finally, in the module/plugin, I have used this snippet to make sure group field data is matched properly:
```
use Craft;
use craft\fields\Matrix;
use verbb\formie\base\Integration;
use verbb\formie\events\ModifyFieldIntegrationValuesEvent;
use yii\base\Event;

Event::on(
  Integration::class,
  Integration::EVENT_MODIFY_FIELD_MAPPING_VALUES,
  function (ModifyFieldIntegrationValuesEvent $event) {
    foreach ($event->fieldSettings as $fieldSetting) {
      if ($fieldSetting->sourceType === Matrix::class) {
        $event->fieldValues[$fieldSetting->handle] = Craft::$app->getRequest()->getBodyParam(
          "fields.{$fieldSetting->handle}",
          $event->fieldValues[$fieldSetting->handle],
        );
      }
    }
  }
);
```

If the group is mapped to the matrix field in the Entry integration in the form settings, the entry is properly saved in the dashboard. Other than that, and based on the link @engram-design provided, you could improvise the values for the entry title and/slug, but those can be also configured in the form settings and mapped via integration config.

I haven't tested this with Super Table or Neo, only with Matrix, but they should work similarly.
